### PR TITLE
Convert DANDI: resource identifiers to lowercase when prepending "https://identifiers.org/" and tighten regex

### PR DESCRIPTION
**Pull request by jwodder**
See https://github.com/dandi/helpdesk/issues/37#issuecomment-962037772

### CODE REVIEW

**Comment by yarikoptic** [comment]
@satra -- do you think it would be a good/kosher thing to do in the long run too?

**Comment by satra** [comment]
> do you think it would be a good/kosher thing to do in the long run too?

yes, unless identifiers.org flips around and creates the opposite error.

**Comment by satra** [comment]
btw, this should trap for draft identifiers. those are not supported by identifiers.org.

**Comment by yarikoptic** [comment]
> btw, this should trap for draft identifiers. those are not supported by identifiers.org.

decision from standup IIRC was: make regex for `DANDI:` identifiers match the one in dandischema which should make it stricter, and proceed.

**Comment by jwodder** [comment]
@yarikoptic @satra When I went to restrict the format of DANDI: identifiers, I found that dandischema and dandi-cli use two different regexes for non-draft version identifiers: dandischema uses `\d+\.\d+\.\d+`, while dandi-cli uses `[.0-9]{5,}`.  Which one should be preferred?

**Comment by satra** [comment]
technically we are using dates and times so ideally: `0\.\d{6}\.\d{4}`, but the current setup is made more general than that with just `\d+` so we don't have to be super strict. also the only thing that can generate this is the API-server on publish, and were reserving the right to alter it.

**Comment by jwodder** [comment]
@satra I'm going to change the regex in dandi-cli to `[0-9]+\.[0-9]+\.[0-9]+`, OK?

**Comment by satra** [comment]
> I'm going to change the regex in dandi-cli to [0-9]+\.[0-9]+\.[0-9]+, OK?

sounds ok but, perhaps just use `\d+`

**Comment by jwodder** [comment]
@satra `\d` accepts Unicode digits outside of `0-9` that we don't want.

File: dandi/dandiarchive.py [hunk 1]
```diff
@@ -363,19 +363,26 @@ class _dandi_url_parser:
         # TODO: Later should better conform to our API, so we could allow
         #       for not only "dandiarchive.org" URLs
         (
-            re.compile(r"DANDI:.*"),
-            {"rewrite": lambda x: "https://identifiers.org/" + x},
-            "DANDI:<dandiset id>",
+            re.compile(
+                fr"DANDI:{DANDISET_ID_REGEX}(?:/[0-9]+\.[0-9]+\.[0-9]+)?",
```
Comment (yarikoptic) [inline dandi/dandiarchive.py:367]: Can't `VERSION_REGEX` be used here instead of duplicating regex?

Comment (jwodder) [reply dandi/dandiarchive.py:367]: @yarikoptic We need to exclude `draft`, which `VERSION_REGEX` includes.

Comment (yarikoptic) [reply dandi/dandiarchive.py:367]: I guess I would have made this one to not include it, and ANY_VERSION_REGEX or alike to include it?

Comment (jwodder) [reply dandi/dandiarchive.py:367]: @yarikoptic Do you want me to do that, or are you concerned about breaking backwards compatibility?

Comment (yarikoptic) [reply dandi/dandiarchive.py:367]: to keep compatible, lets go the other way: add `PUBLISHED_VERSION_REGEX = '[0-9]+\.[0-9]+\.[0-9]+'`?

Comment (jwodder) [reply dandi/dandiarchive.py]: @yarikoptic Done.

**Comment by yarikoptic** [comment]
sorry, forgot about this one. Thank you @jwodder.   Let's proceed -- I think it would make the world a better place ;)  I will even release with it

**Pull request merged**